fe/src/components/Issues/IssueDetail/IssueDetailHeader.tsx

```diff
@@ -0,0 +1,214 @@
+import alertIcon from "@assets/icon/alertCircle.svg";
+import archiveIcon from "@assets/icon/archive.svg";
+import editIcon from "@assets/icon/edit.svg";
+import xIcon from "@assets/icon/xSquare.svg";
+import Button from "@components/common/Button";
+import TextInput from "@components/common/TextInput";
+import { IssueDetails } from "@customTypes/index";
+import { convertPastTimestamp } from "@utils/time";
+import { putIssueIsOpen, putIssueTitle } from "api";
+import { FormEvent, useEffect, useState } from "react";
+import { styled } from "styled-components";
+
+export default function IssueDetailHeader({
+  issueDetails,
+  updateIssueTitle,
+  updateIssueIsOpen,
+  numComments,
+}: {
+  issueDetails: IssueDetails | null;
+  updateIssueTitle: (newTitle: string) => void;
+  updateIssueIsOpen: () => void;
+  numComments: number;
+}) {
+  const [isEditTitle, setIsEditTitle] = useState(false);
+  const [newTitle, setNewTitle] = useState("");
+
+  const { issueId, author, title, createdAt, isOpen } = issueDetails || {
+    issueId: 0,
+    author: { username: "", profileURl: "" },
+    title: "",
+    createdAt: new Date().toISOString(),
+    isOpen: true,
+  };
+
+  useEffect(() => {
+    setNewTitle(title);
+  }, [title]);
```

Comment:
issueDetails fetch 완료되면 기존 title을 newTitle로 설정하기 위한 것 같은데..
useEffect 없이 issueDetails prop이 변경되면 IssueDetailHeader 렌더링 되도록 하는 방법이 없을까요? 저도 아직 정확한 이유는 정확히 모르겠지만, 공식문서에서 useEffect로 자식의 state를 set하지 말라고 나와있어서 공유드려요.

### 참고
> [prop이 변경되면 모든 state 재설정하기](https://react-ko.dev/learn/you-might-not-need-an-effect#resetting-all-state-when-a-prop-changes)
> [prop 변경에 대한 응답으로 특정 state 일부를 조정하려면 렌더링 중에 설정하세요.](https://react-ko.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes)

Reply:
공식문서에서 이 방식이 안좋다고 말한 이유는 prop을 받는 자식 컴포넌트와 그 안에 있는 컴포넌트들이 처음 렌더할 때 stale한 값을 먼저 렌더하고 그 다음에 다시 올바른 값으로 렌더를 하게 되기 때문이라고 하네요.
> This is inefficient because ProfilePage and its children will first render with the stale value, and then render again.